lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1826,7 +1822,8 @@ void InferenceMapping::constrainTypes(Value larger, Value smaller) {
         }
       };
 
-  constrain(getBaseType(type), larger, smaller);
+  if (auto type = larger.getType().dyn_cast<FIRRTLType>())
```

Comment:
The pattern here was to check the result of `getBaseType()` but here it's used unchecked.  Should this also check result of getBaseType?

Alternatively `getBaseType()` could support a null or even non-FIRRTL type now that the return value needs to be checked already anyway.

Reply:
Went with `getBaseType` accepts `Type` instead of `FIRRTLType`, and thus returns null when passed a foreign type. https://github.com/llvm/circt/pull/5392